src/services/newMessagesService.ts

```diff
@@ -22,26 +25,58 @@ import { pipe } from "fp-ts/lib/function";
 import * as E from "fp-ts/Either";
 import * as O from "fp-ts/Option";
 import * as T from "fp-ts/Task";
-import { PaginatedPublicMessagesCollection } from "../../generated/io-api/PaginatedPublicMessagesCollection";
+import * as TE from "fp-ts/TaskEither";
+import { IResponseType } from "@pagopa/ts-commons/lib/requests";
+
+import { InternalMessageResponseWithContent } from "../../generated/io-messages-api/InternalMessageResponseWithContent";
+import { CreatedMessageWithContent } from "../../generated/io-messages-api/CreatedMessageWithContent";
+import { LegalData } from "../../generated/io-messages-api/LegalData";
+import { LegalMessageWithContent } from "../../generated/backend/LegalMessageWithContent";
+import { PaginatedPublicMessagesCollection } from "../../generated/io-messages-api/PaginatedPublicMessagesCollection";
 import { GetMessageParameters } from "../../generated/parameters/GetMessageParameters";
 import { GetMessagesParameters } from "../../generated/parameters/GetMessagesParameters";
 
 import { CreatedMessageWithContentAndAttachments } from "../../generated/backend/CreatedMessageWithContentAndAttachments";
 import { getPrescriptionAttachments } from "../utils/attachments";
+import { StrictUTCISODateFromString } from "../utils/date";
+import { errorsToError } from "../utils/errorsFormatter";
+import { PecBearerGeneratorT } from "../types/token";
 import { User } from "../types/user";
 import {
+  IResponseSuccessOctet,
   ResponseErrorStatusNotDefinedInSpec,
   ResponseErrorUnexpectedAuthProblem,
+  ResponseSuccessOctet,
   unhandledResponseStatus,
   withCatchAsInternalError,
-  withValidatedOrInternalError
+  withValidatedOrInternalError,
+  wrapValidationWithInternalError
 } from "../utils/responses";
 import { MessageStatusChange } from "../../generated/io-messages-api/MessageStatusChange";
 import { MessageStatusAttributes } from "../../generated/io-messages-api/MessageStatusAttributes";
 
+import { LegalMessage } from "../../generated/pecserver/LegalMessage";
+import { IPecServerClientFactoryInterface } from "./IPecServerClientFactory";
+
+const isGetMessageSuccess = (
+  res: IResponseType<number, unknown, never>
+): res is IResponseType<200, InternalMessageResponseWithContent, never> =>
+  res.status === 200;
+
+const isPecServerGetMessageSuccess = (
+  res: IResponseType<number, unknown, never>
+): res is IResponseType<200, LegalMessage, never> => res.status === 200;
+
+const MessageWithLegalData = t.intersection([
+  CreatedMessageWithContent,
+  t.interface({ content: t.interface({ legal_data: LegalData }) })
+]);
+type MessageWithLegalData = t.TypeOf<typeof MessageWithLegalData>;
+
 export default class NewMessagesService {
```

Comment:
This service could be renamed to `MessageService` since the old one has been purged from the code. Every `newMessagesService` variable could be renamed too.
Personally I would avoid naming things with "New" because if a day there will be another "New" services it should be named "Newest" or "NewNew" 😁 .

Reply:
Since it's already a huge PR, can we do in another one?

Reply:
For me it's ok, we can just track this on a ticket. :)